src/lib/components/selectv2/Selectv2.component.js

```diff
@@ -0,0 +1,280 @@
+    'aria-selected': props.isSelected,
+  };
+
+  return (
+    <components.Option
+      {...props}
+      innerProps={innerProps}
+      isFocused={props.isFocused && props.selectProps.keyboardFocusEnabled}
+    >
+      <div className="option-value-wrapper">
+        <div className="option-icon">{props.data.icon}</div>
+        {formatOptionLabel()}
+      </div>
+      <div>{props.isDisabled && <i className="fas fa-ban" />}</div>
+    </components.Option>
+  );
+};
+
+const Menu = (props) => {
+  useEffect(() => {
+    props.selectProps.setIsMenuBottom(props.placement === 'bottom');
+  }, [props]);
+  return <components.Menu {...props} />;
+};
+
+const ValueContainer = ({ children, ...props }) => {
+  const selectedOption = props.selectProps.selectedOption;
+  const icon = selectedOption ? selectedOption.icon : null;
+  return (
+    <components.ValueContainer {...props}>
+      {icon ? <div className="value-container-icon">{icon}</div> : null}
+      <div>{children}</div>
+    </components.ValueContainer>
+  );
+};
+
+export type SelectProps = {
+  placeholder?: string,
+  disabled?: boolean,
+  children: Node,
+  defaultValue?: string,
+  value?: string,
+  onChange: (newValue: string) => void,
+  variant?: 'default' | 'rounded',
+  className?: string,
+};
+
+type SelectOptionProps = {
+  value: string,
+  label: Node,
+  isDisabled: boolean,
+  icon?: Element<typeof Icon>,
+  optionProps: any,
+};
+
+// more/equal than NOPT_SEARCH options enable search
+const NOPT_SEARCH = 8;
+
+function SelectBox({
+  placeholder = 'Select...',
+  disabled = false,
+  children,
+  defaultValue,
+  value,
+  onChange,
+  variant = 'default',
+  className,
+  ...rest
+}: SelectProps) {
+  const [keyboardFocusEnabled, setKeyboardFocusEnabled] = useState(false);
+  const [searchSelection, setSearchSelection] = useState('');
+  const [searchValue, setSearchValue] = useState('');
+  const [customPlaceholder, setPlaceholder] = useState(placeholder);
+  const [options, setOptions] = useState<Array<SelectOptionProps> | null>(null);
+  const isDefaultVariant = variant === 'default';
+  const [isMenuBottom, setIsMenuBottom] = useState(true);
+  const [selectRef, setSelectRef] = useState(null);
+
+  const childrenToOptions = useCallback((): Array<SelectOptionProps> => {
+    if (children) {
+      return React.Children.toArray(children)
+        .filter((child) => child.type === Option)
+        .map((child) => {
+          const {
+            value,
+            children,
+            disabled,
+            icon,
+            ...rest
+          }: OptionProps = child.props;
+          return {
+            value: value,
+            label: children,
+            isDisabled: disabled || false,
+            icon: icon,
+            optionProps: { ...rest },
+          };
+        });
+    } else {
+      return [];
+    }
+  }, [children]);
+
+  useEffect(() => {
+    setOptions(childrenToOptions());
+  }, [childrenToOptions]);
```

Comment:
I think this can be refactored to a simple useMemo with children as a dependency array. It will avoid an unnecessary render linked to the useEffect 

```suggestion
  const options = useMemo((): Array<SelectOptionProps> => {
    if (children) {
      return React.Children.toArray(children)
        .filter((child) => child.type === Option)
        .map((child) => {
          const {
            value,
            children,
            disabled,
            icon,
            ...rest
          }: OptionProps = child.props;
          return {
            value: value,
            label: children,
            isDisabled: disabled || false,
            icon: icon,
            optionProps: { ...rest },
          };
        });
    } else {
      return [];
    }
  }, [children]);
```

Then the state options could be dropped.